calico_kubernetes/policy.py

```diff
@@ -0,0 +1,224 @@
+        """
+        Generates a libcalico
+        :param arguments: A dictionary of arguments with the following format.
+
+        {
+            "<DSTCIDR>": null,
+            "<DSTPORTS>": null,
+            "<DSTTAG>": null,
+            "<ICMPCODE>": null,
+            "<ICMPTYPE>": null,
+            "<SRCCIDR>": null,
+            "<SRCPORTS>": "80",
+            "<SRCTAG>": null,
+            "allow": true,
+            "cidr": 0,
+            "code": false,
+            "deny": false,
+            "from": true,
+            "icmp": false,
+            "ports": 1,
+            "tag": 0,
+            "tcp": true,
+            "to": false,
+            "type": false,
+            "udp": false
+        }
+
+        :return: A libcalico Rule object
+        """
+        # We only support whitelist rules.
+        rule_args = {"action": "allow"}
+
+        # Get arguments.
+        if arguments.get("tcp"):
+            protocol = "tcp"
+        elif arguments.get("udp"):
+            protocol = "udp"
+        elif arguments.get("icmp"):
+            protocol = "icmp"
+        else:
+            protocol = None
+
+        src_ports = arguments.get("<SRCPORTS>")
+        dst_ports = arguments.get("<DSTPORTS>")
+        icmp_type = arguments.get("<ICMPTYPE>")
+        icmp_code = arguments.get("<ICMPCODE>")
+        src_net = arguments.get("<SRCCIDR>")
+        src_label = arguments.get("<SRCLABEL>")
+        dst_net = arguments.get("<DSTCIDR>")
+        dst_label= arguments.get("<DSTLABEL>")
+
+        # Populate rule arguments
+        if protocol:
+            rule_args["protocol"] = protocol
+        if src_ports:
+            rule_args["src_ports"] = self._validate_ports(src_ports)
+        if dst_ports:
+            rule_args["dst_ports"] = self._validate_ports(dst_ports)
+        if icmp_type:
+            rule_args["icmp_type"] = self._validate_int(icmp_type)
+        if icmp_code:
+            rule_args["icmp_code"] = self._validate_int(icmp_code)
+        if src_net:
+            rule_args["src_net"] = self._validate_ip(src_net)
+        if src_label:
+            rule_args["src_tag"] = self._validate_label(src_label)
+        if dst_net:
+            rule_args["dst_net"] = self._validate_ip(dst_net)
+        if dst_label:
+            rule_args["dst_tag"] = self._validate_label(dst_label)
+
+        return Rule(**rule_args)
+
+    def _validate_int(self, value):
+        """
+        Validates the given value can be parsed as an int.
+        """
+        try:
+            return int(value)
+        except ValueError:
+            raise PolicyParseError("Failed to parse %s, expecting int", value)
+
+    def _validate_label(self, value):
+        """
+        Takes the given label, validates it, and returns the equivalent tag.
+        """
+        if not "=" in value:
+            raise PolicyParseError("Failed to parse %s, expecting "
+                                   "label of form X=Y", value)
+        k,v = value.split("=")
```

Comment:
This isn't going to render `a=b=c` correctly. Can we insert this into the docopt parse string? If you can't do arbitrary k=v there, then I'd prefer you did a regex here so that we know there's only one `=`.